lib/services/project-service.ts

```diff
@@ -56,20 +54,25 @@ export class ProjectService implements IProjectService {
 		const { template, projectDir, appId, projectName, ignoreScripts } = projectCreationSettings;
 
 		try {
-			const { templatePath, templateVersion } = await this.$projectTemplatesService.prepareTemplate(template, projectDir);
-			await this.extractTemplate(projectDir, templatePath, templateVersion);
+			const templateData = await this.$projectTemplatesService.prepareTemplate(template, projectDir);
+			const templatePackageJsonContent = templateData.templatePackageJsonContent;
+			const templateVersion = templateData.templateVersion;
 
-			await this.ensureAppResourcesExist(projectDir);
+			await this.extractTemplate(projectDir, templateData);
+
+			if (templateVersion === constants.TemplateVersions.v2) {
+				this.alterPackageJsonData(projectDir, appId);
+			}
 
-			const templatePackageJsonData = this.getDataFromJson(templatePath);
+			await this.ensureAppResourcesExist(projectDir);
 
-			if (!(templatePackageJsonData && templatePackageJsonData.dependencies && templatePackageJsonData.dependencies[constants.TNS_CORE_MODULES_NAME])) {
+			if (!(templatePackageJsonContent && templatePackageJsonContent.dependencies && templatePackageJsonContent.dependencies[constants.TNS_CORE_MODULES_NAME])) {
 				await this.$npmInstallationManager.install(constants.TNS_CORE_MODULES_NAME, projectDir, { dependencyType: "save" });
```

Comment:
Maybe we can improve this just to persist the value in package.json instead of executing `npm install`, but this could be done in a separate PR.